### PR TITLE
WIP: Tigervnc

### PR DESCRIPTION
**Pull request by manics**
This is a prerequisite for https://github.com/conda-forge/staged-recipes/pull/21369
This is a modified version of https://github.com/conda-forge/staged-recipes/pull/17740

Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by manics** [comment]
Still todo:
- Fix tests (`--help` runs but returns exit code 1)
- Add `Xvnc` and `libvnc.so`: https://github.com/TigerVNC/tigervnc/blob/v1.12.0/BUILDING.txt#L100-L146

**Comment by manics** [comment]
This fails when trying to build Xvnc:
```
checking for fixesproto >= 5.0 damageproto >= 1.1 xcmiscproto >= 1.2.0 xtrans >= 1.3.5 bigreqsproto >= 1.1.0 xproto >= 7.0.31 randrproto >= 1.6.0 renderproto >= 0.11 xextproto >= 7.2.99.901 inputproto >= 2.3 kbproto >= 1.0.3 fontsproto >= 2.1.3 pixman-1 >= 0.27.2 videoproto compositeproto >= 0.4 recordproto >= 1.13.99.1 scrnsaverproto >= 1.1 resourceproto >= 1.2.0 presentproto >= 1.1 xkbfile  pixman-1 >= 0.27.2 xfont2 >= 2.0.0 xau xdmcp... no
configure: error: Package requirements (fixesproto >= 5.0 damageproto >= 1.1 xcmiscproto >= 1.2.0 xtrans >= 1.3.5 bigreqsproto >= 1.1.0 xproto >= 7.0.31 randrproto >= 1.6.0 renderproto >= 0.11 xextproto >= 7.2.99.901 inputproto >= 2.3 kbproto >= 1.0.3 fontsproto >= 2.1.3 pixman-1 >= 0.27.2 videoproto compositeproto >= 0.4 recordproto >= 1.13.99.1 scrnsaverproto >= 1.1 resourceproto >= 1.2.0 presentproto >= 1.1 xkbfile  pixman-1 >= 0.27.2 xfont2 >= 2.0.0 xau xdmcp) were not met:

Requested 'randrproto >= 1.6.0' but version of RandrProto is 1.5.0
No package 'xkbfile' found
No package 'xfont2' found
```
I can't see a way forward from here.

**Comment by weiji14** [comment]
> Requested 'randrproto >= 1.6.0' but version of RandrProto is 1.5.0

Need to bump up the version at https://github.com/conda-forge/xorg-randrproto-feedstock/blob/0af6e0d5af5e9dc3bf71bb59d5e782479dbc0032/recipe/meta.yaml#L4 from 1.5.0 to 1.6.0?

> No package 'xkbfile' found
> No package 'xfont2' found

Not sure about these, don't see them on conda-forge yet.

File: recipes/tigervnc/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+requirements:
+  build:
+    - cmake
+    - {{ compiler('cxx') }}
+    - make
+    # For building Xvnc
+    - autoconf
+    - automake
+    - patch
+    - libtool
```
Comment (weiji14) [inline recipes/tigervnc/meta.yaml:34]: > No package 'xkbfile' found

Actually, just found https://anaconda.org/conda-forge/libxkbfile, one step closer I guess

```suggestion
    - libtool
    - libxkbfile
```

**Comment by manics** [comment]
It looks like randrproto isn't pulling from a standard source repo, which is probably why it's not been automatically updated:
https://github.com/conda-forge/xorg-randrproto-feedstock/blob/0af6e0d5af5e9dc3bf71bb59d5e782479dbc0032/recipe/meta.yaml#L4-L14

**Comment by weiji14** [comment]
> It looks like randrproto isn't pulling from a standard source repo, which is probably why it's not been automatically updated: https://github.com/conda-forge/xorg-randrproto-feedstock/blob/0af6e0d5af5e9dc3bf71bb59d5e782479dbc0032/recipe/meta.yaml#L4-L14

Was gonna manually bump randrproto from 1.5.0 to 1.6.0, but I couldn't find it in https://www.x.org/releases/individual/proto or https://gitlab.freedesktop.org/xorg/proto/randrproto, the latest tagged version seems to be 1.5.0?!! No idea where tigervnc is pulling this from :sweat_smile:

File: recipes/tigervnc/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    - xorg-damageproto
+    - xorg-fontsproto
+    - xorg-glproto
+    - xorg-presentproto
+    - xorg-randrproto
+    - xorg-recordproto
+    - xorg-resourceproto
+    - xorg-scrnsaverproto
+    - xorg-videoproto
+    - xorg-xcmiscproto
```
Comment (weiji14) [inline recipes/tigervnc/meta.yaml:51]: According to https://github.com/conda-forge/staged-recipes/pull/14990#issue-897301389, all the X.org "proto" packages has been combined to `xorgproto`. Not sure if this works, but could test it out.

```suggestion
    - xorg-xorgproto
```

File: recipes/tigervnc/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
```
Comment (weiji14) [inline recipes/tigervnc/meta.yaml:76]:
```suggestion
    - xorg-xorgproto
```

**Comment by weiji14** [comment]
@conda-forge-admin, please restart ci

File: recipes/tigervnc/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,92 @@
+{% set name = "tigervnc" %}
+{% set version = "1.12.0" %}
```
Comment (weiji14) [inline recipes/tigervnc/meta.yaml:2]: Try building [tigervnc=1.13.1](https://github.com/TigerVNC/tigervnc/releases/tag/v1.13.1)?

```suggestion
{% set version = "1.13.1" %}
```

File: recipes/tigervnc/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+source:
+  - url: https://github.com/TigerVNC/tigervnc/archive/v{{ version }}.tar.gz
+    sha256: 9ff3f3948f2a4e8cc06ee598ee4b1096beb62094c13e0b1462bff78587bed789
+    patches:
+      # Disables vncsession (requires PAM and runs as root)
+      # Disable PAM password check (always return failure)
+      - ignore-pam.patch
+  # https://github.com/TigerVNC/tigervnc/blob/v1.12.0/BUILDING.txt#L100-L146
+  - url: https://github.com/freedesktop/xorg-xserver/archive/xorg-server-{{ xorg_x11_server_version }}.tar.gz
+    sha256: 68105526da70d04f7111339044cdd5b2a762717178cd345aba6cb2d1b399c71b
```
Comment (weiji14) [inline recipes/tigervnc/meta.yaml:18]: Seems like xorg-xserver has moved from GitHub to GitLab at https://gitlab.freedesktop.org/xorg/xserver? The GitHub link doesn't work anymore

**Pull request closed without merging**